InfraDNS/Build.ps1

```diff
@@ -44,30 +46,30 @@ Task InstallModules -Depends GenerateEnvironmentFiles {
     # Install resources on build agent
     "Installing required resources..."
 
-    #Workaround for bug in Install-Module cmdlet
-    if(!(Get-PackageProvider -Name NuGet -ListAvailable -ErrorAction Ignore))
+    # Workaround for bug in Install-Module cmdlet
+    if(!(Get-PackageProvider -Name NuGet -ListAvailable -ErrorAction  Continue ))
     {
         Install-PackageProvider -Name NuGet -Force
     }
     
-    if (!(Get-PSRepository -Name PSGallery -ErrorAction Ignore))
+    if (!(Get-PSRepository -Name PSGallery -ErrorAction Continue))
     {
         Register-PSRepository -Name PSGallery -SourceLocation https://www.powershellgallery.com/api/v2/ -InstallationPolicy Trusted -PackageManagementProvider NuGet
     }
     
-    #End Workaround
+    # End Workaround
     
     foreach ($Resource in $RequiredModules)
     {
-        Install-Module -Name $Resource.Name -RequiredVersion $Resource.Version -Repository 'PSGallery' -Force
-        Save-Module -Name $Resource.Name -RequiredVersion $Resource.Version -Repository 'PSGallery' -Path $ModuleArtifactPath -Force
+        Install-Module -Name $Resource.Name -RequiredVersion $Resource.Version -Force
```

Comment:
I found the opposite to be true. It wouldn't run with PSGallery explicitly set as the Repository, but did fine when it wasn't. I rat-holed on that for a long time.